### ISSUE TITLE
Want to update the keras model with different losses by train_on_batch methods. I have given the pytorch version code which is perfectly working. Please have a look

### ISSUE THREAD

**Issue by TanmDL** [keras-team/keras#14611]
def MLP():

    ip1=Input(shape=(512,))

    x=Dense(256)(ip1)
    x=LeakyReLU(alpha=0.1)(x)

    x=Dense(128)(x)
    x=LeakyReLU(alpha=0.1)(x)

    Final=Dense(10, activation='softmax')(x)

    Mlp=Model(ip1, Final)
    Mlp.summary()
    return Mlp

##### define two demo loss functions here.

loss1 = RMSE()
loss2 = MSE()

mlp1=MLP() 
mlp1.compile(loss=loss1, optimizer=adamOpt)

mlp2=MLP()
mlp2.compile(loss=loss2, optimizer=adamOpt)'

##### Now I want to train the model 

for (i, data, label) in enumerate(dataloader):

    mlp1.train_on_batch(data[i], label[i])

    mlp2.train_on_batch(data[i], label[i])

#################### the pytorch version code is given below
Pytorch_model = MLP() (The same model which is written in Pytorch)

loss1 = RMSE()
loss2 = MSE()

optim_loss1 = torch.optim.Adam(Pytorch_model.parameters(), lr = 0.2)
optim_loss2 = torch.optim.Adam(Pytorch_model.parameters(), lr = 0.2)

for (i, data, label) in enumerate(dataloader):

    Pytorch_model.zero_grad()
    Pytorch_model.train()
    optim_loss1.zero_grad()
    esti_out = Pytorch_model(data)
    1st_loss = loss1(esti_out, label)
    1st_loss.backward()
    optim_loss1.step()

here, 2nd time update the same model parameters with loss2

    optim_loss2.zero_grad()
    esti_out2 = Pytorch_model(data)
    2nd_loss = loss2(esti_out2, label)
    2nd_loss.backward()
    optim_loss2.step()

 In pytorch,  the MLP weight is updated perfectly by two losses but I don't know the Keras version is perfectly written or not.

**Comment by sushreebarsa** [comment]
@TanmDL In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!
